monitor_state_dict.py

```diff
@@ -0,0 +1,110 @@
+"""
+Contains the definitions for monitor.state.
+
+This is essentially a stopgap while OOP state is worked on.
+"""
+from __future__ import annotations
+
+from typing import MutableMapping, TYPE_CHECKING, Any, DefaultDict, Dict, List, Literal, Optional, Set, Tuple, TypedDict
+
+
+class MonitorStateDict(TypedDict):
+    """Top level state dictionary for monitor.py."""
+
+    # Game related
+    GameLanguage:           Optional[str]               # From `Fileheader`
+    GameVersion:            Optional[str]               # From `Fileheader`
+    GameBuild:              Optional[str]               # From `Fileheader`
+    Horizons:               Optional[bool]                        # Does the player have Horizons?
+    Odyssey:                bool                        # Have we detected Odyssey?
+
+    # Multi-crew
+
+    Captain:                Optional[str]               # If on a crew, the captian's name
+    Role:                   Optional[Literal['Idle', 'FireCon', 'FighterCon']]  # Role in crew
+
+    # Cmdr state
+    FID:                    Optional[str]               # Frontier CMDR ID
+    Friends:                Set[str]                    # Online Friends
+    Credits:                int
+    Loan:                   Optional[int]
+    Engineers:              Dict[Any, Any]              # TODO
+    Rank:                   Dict[Any, Any]              # TODO
+    Reputation:             Dict[Any, Any]              # TODO
+    Statistics:             Dict[Any, Any]              # This is very freeform.
+
+    # Engineering Materials
+    Raw:                    DefaultDict[str, int]
+    Encoded:                DefaultDict[str, int]
+    Manufactured:           DefaultDict[str, int]
+
+    # Ship
+    ShipID:                 Optional[str]
+    ShipIdent:              Optional[str]
+    ShipName:               Optional[str]
```

Comment:
ShipName could be problematic.  Check the multi-line comment at monitor.py:596 (`event_type == 'loadout'` code).

Although given `''` is falsey that code doesn't actually need to check for that after  the `if entry['ShipName']`, so perhaps it will be OK.

Reply:
Direct checks for `ShipName` appear to always do a falsey check so it should be fine -- obviously this entire PR will need testing